Feature/update dependencies

davidalbela opened the pull request:
Update contracts to Solidity `v0.5.0` and update nodeJs dependencies:
- Update node dependencies.
- Update truffle configuration with new HDWalletProvider.
- Update Solidity to `v0.5.0`.
- Update Rinkeby contract address.
- Update Travis CI file.

cag commented on truffle.js:
```diff
@@ -1,34 +1,30 @@
     development: {
       host: "127.0.0.1",
-      port: 9545,
+      port: 8545,
```
Huh weird, I wonder why I had this...

Anyway, +1 to changing it back to the standard port.

cag commented on contracts/TokenSplitter.sol:
```diff
@@ -1,10 +1,16 @@
 contract TokenSplitter {
-    function splitTokens(address[] _recipients, address _tokenAddress, uint _amountForEach) public {
+
+    uint overflowed = 2**32;
```
Why the overflow limit?

cag commented on contracts/TokenSplitter.sol:
```diff
@@ -1,10 +1,16 @@
+    uint overflowed = 2**32;
+
+    function splitTokens(address[] memory _recipients, address _tokenAddress, uint _amountForEach) public {
+        require(_recipients.length < 21, "TokenSplitter: only 20 addresses allowed");
```
Why only 20 addresses max?

davidalbela commented:
> In general, I'd be fine with updating dependencies, except this PR does more than that. I'm not against the changes per se, but I don't see a reason necessarily for adding them either.
> 
> It seems a recipient cap would be enforced by gas limits anyway, so I don't see a reason to limit this to 20 per transaction if the blocks can accommodate more.
> 
> As for the overflow limit, 2**32 would be way too low anyway (that's about 4.3 GWei). Why not let the ERC20 token itself enforce that limit? It would have to keep track of balances anyway.

Ok, removed the `require` functions from contract. Sorry, I should not add that to that commit neither the PR.

Updated Travis CI file as well :)

cag commented on .travis.yml:
```diff
@@ -1,12 +1,12 @@
   - truffle compile
 script:
-  - ganache-cli -d -p 9545 --defaultBalanceEther 10000 -a 30 > /dev/null &
+  - ganache-cli -d -p 8545 --defaultBalanceEther 10000 -a 30 > /dev/null &
```
Could even remove `-p 8545` now since that's the default, but anyway that's not really a big deal.
